.github/workflows/run_test_windows.yml

```diff
@@ -0,0 +1,287 @@
+                  AZLE_IS_FEATURE_BRANCH_PR="${{ steps.set-conditions.outputs.is_feature_branch_pr || 'false' }}"
+                  AZLE_IS_FEATURE_BRANCH_DRAFT_PR="${{ steps.set-conditions.outputs.is_feature_branch_draft_pr || 'false' }}"
+
+                  echo "AZLE_IS_MAIN_BRANCH_PUSH_FROM_FEATURE_MERGE: $AZLE_IS_MAIN_BRANCH_PUSH_FROM_FEATURE_MERGE"
+                  echo "AZLE_IS_MAIN_BRANCH_PUSH_FROM_RELEASE_MERGE: $AZLE_IS_MAIN_BRANCH_PUSH_FROM_RELEASE_MERGE"
+                  echo "AZLE_IS_RELEASE_BRANCH_PR: $AZLE_IS_RELEASE_BRANCH_PR"
+                  echo "AZLE_IS_FEATURE_BRANCH_PR: $AZLE_IS_FEATURE_BRANCH_PR"
+                  echo "AZLE_IS_FEATURE_BRANCH_DRAFT_PR: $AZLE_IS_FEATURE_BRANCH_DRAFT_PR"
+              shell: wsl-bash {0}
+
+            - name: Report full path of test
+              # Just in case the path isn't obvious from the name, this will remove ambiguity
+              run: echo ${{matrix.test.path}}
+              shell: wsl-bash {0}
+
+            - name: Setup Node.js in WSL
+              run: |
+                  # Use clean PATH to avoid Windows conflicts
+                  export PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
+
+                  # Get Node version from package.json
+                  NODE_VERSION=$(jq -r '.azle.globalDependencies.node // error("node version not found")' "package.json")
+                  # Extract major version number (e.g., "22.14.0" -> "22")
+                  MAJOR_VERSION=$(echo $NODE_VERSION | cut -d. -f1)
+                  echo "Installing Node.js version $NODE_VERSION (major version $MAJOR_VERSION) in WSL..."
+
+                  # Install Node.js using NodeSource repository
+                  curl -fsSL https://deb.nodesource.com/setup_${MAJOR_VERSION}.x | sudo -E bash -
+                  sudo apt-get install -y nodejs
+              shell: wsl-bash {0}
+
+            - name: Install dfx
+              run: |
+                  # Use clean PATH to avoid Windows conflicts
+                  export PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
+
+                  DFX_VERSION=$(jq -r '.azle.globalDependencies.dfx // error("dfx version not found")' "package.json")
+                  echo "Installing dfx version $DFX_VERSION..."
+                  DFXVM_INIT_YES=true DFX_VERSION=$DFX_VERSION sh -ci "$(curl --retry 3 -fsSL https://sdk.dfinity.org/install.sh)"
+                  export PATH="$HOME/.local/share/dfx/bin:$PATH"
+                  if ! command -v dfx &> /dev/null; then
+                      echo "dfx installation verification failed" >&2
+                      exit 1
+                  fi
+                  echo "dfx installation completed."
+              shell: wsl-bash {0}
+
+            - name: Install Rust and Cargo
+              run: |
+                  # Use clean PATH to avoid Windows conflicts
```

Comment:
We aren't using the retry command action

Reply:
There are a number of weird things about working in WSL and if we just have everything local to the wsl workflow in one big beautiful step then we 1) localize all of the WSL weirdness to a single file, and 2) mitigate a lot of the weirdness. The steps will need to be completely separately so we will have the same amount of code duplication. The difference is where. I think it is more advantageous to have all of the WSL stuff together than all of the retry stuff together. And I'm not even sure that making the retry action os agnostic would even work. And we have already spent way too long on this project